### DISCUSSION TITLE
ESP12F Instead?

### DISCUSSION THREAD

**Discussion by jhogendorn** [CircuitSetup/Expandable-6-Channel-ESP32-Energy-Meter#103] [Q&A]
Is it possible to interface in an ESP12F board, instead of an ESP32?

These relay boards are very useful, i have a few hanging around for potential projects: https://www.aliexpress.com/item/1005002350865166.html

I've had a (very) quick look at the schematic and it seems its just passing power and SPI pins for comms? If thats the case if I make an appropriate pigtail/adapter board it could work?

Or is it smarter to have a dumb relay board and pull gpio off the esp32?

**Comment by CircuitSetup** [comment]
Depending on what software you want to run on the ESP12F, it could work, yes. You would just need to hook up the 3.3V, GND, MOSI, MISO, and CS pins. Other GPIO pins are hooked up, but are not used. I haven't had the opportunity to test this though, so it may not have enough processing power to handle WiFi, SPI, and whatever switching you want to do. 

It may be easier to use an ESP32, and use a dumb relay board, as you mentioned.

**Reply by jhogendorn** [reply]
Ok, thats promising. I was hoping to run ESPHome. (https://www.esphome-devices.com/devices/ESP-12F-Relay-X4)

I will make an attempt and report back.
